### DISCUSSION TITLE
Does ccache support Fortran 77?

### DISCUSSION THREAD

**Discussion by boegel** [ccache/ccache#131] [Questions & answers] [answered]
The release notes for `ccache` 3.3 mention "Added support for Fortran 77".

Is there some more information available about that somewhere, i.e. what is supported exactly, etc.?

It seems like the documentation should be updated as well in some places to reflect that e.g. `gfortran` should be symlinked as well, e.g. https://ccache.samba.org/manual.html#_run_modes ?

How does the current Fortran support compare with the `f90cache` fork (cfr. http://people.irisa.fr/Edouard.Canot/f90cache/), which doesn't seem to support GCC 5.x (and beyond)?

**Comment by jrosdahl** [comment, marked as answer]
> Is there some more information available about that somewhere, i.e. what is supported exactly, etc.?

See c0875cf2dae56f66076ee01e0ca6e723ba999199 and the discussion in #90.

> It seems like the documentation should be updated as well in some places to reflect that e.g. gfortran should be symlinked as well, e.g. https://ccache.samba.org/manual.html#_run_modes ?

That section is not supposed to be a full list of possible compiler names, only an example ("do something like this"), so I don't think we should strive to make it a full list since that will be very long and hard to maintain.

> How does the current Fortran support compare with the f90cache fork (cfr. http://people.irisa.fr/Edouard.Canot/f90cache/), which doesn't seem to support GCC 5.x (and beyond)?

I have no idea, sorry.

**Comment by boegel** [comment]
@jrosdahl Thanks for the info. Your motivation to not update the documentation makes sense.

W.r.t. comparison with `f90cache`: obviously `f90cache` has some support for Fortran90, so that's one key difference. Another is that `f90cache` doesn't seem to support gfortran v5.x or newer. In fact, it's actually pretty harsh there, since it resorts to `exit 1` in case the compiler version is detected as too recent, rather than opting out and acting as a fall-through to the actual compiler (which makes it pretty useless in my view). :(

I'll close this issue since you provided a good reference to the current state of Fortran support, thanks!
